python/cudf/cudf/core/dataframe.py

```diff
@@ -1854,86 +1854,75 @@ def _make_operands_and_index_for_binop(
         ],
         Optional[BaseIndex],
     ]:
-        lhs, rhs = self, other
-
-        if _is_scalar_or_zero_d_array(rhs):
-            rhs = [rhs] * lhs._num_columns
-
-        # For columns that exist in rhs but not lhs, we swap the order so that
-        # we can always assume that left has a binary operator. This
-        # implementation assumes that binary operations between a column and
-        # NULL are always commutative, even for binops (like subtraction) that
-        # are normally anticommutative.
-        # TODO: The above should no longer be necessary once we switch to
-        # properly invoking the operator since we can then rely on reflection.
-        if isinstance(rhs, Sequence):
-            # TODO: Consider validating sequence length (pandas does).
-            operands = {
-                name: (left, right, reflect, fill_value)
-                for right, (name, left) in zip(rhs, lhs._data.items())
-            }
-        elif isinstance(rhs, DataFrame):
+        # Check built-in types first for speed.
+        if isinstance(other, (list, dict, Sequence, Mapping)):
```

Comment:
To be clear, we are not removing support for operations like `DataFrame * 10` right? This is just deprecating for list-like : `DataFrame * [10, 11, 12]`. Correct?

Reply:
Correct. Just sequences. Users can still reproduce this if they want to by converting to a Series with the appropriate index, which will make it obvious that what they're doing requires GPU memory allocations:
```
>>> df = cudf.DataFrame({'a': [1], 'b': [2]})
>>> df + [10, 100]
    a    b
0  11  102
>>> df + cudf.Series({'a': 10, 'b': 100})
    a    b
0  11  102
>>> df + cudf.Series([10, 100], index=['a', 'b'])
    a    b
0  11  102
```